boefjes/boefjes/clients/scheduler_client.py

```diff
@@ -66,20 +78,24 @@ def __init__(self, base_url: str):
     def _verify_response(response: Response) -> None:
         response.raise_for_status()
 
-    def get_queues(self) -> list[Queue]:
-        response = self._session.get("/queues")
+    def pop_item(self, scheduler_id: str) -> Task | None:
+        response = self._session.post(f"/schedulers/{scheduler_id}/pop?limit=1")
         self._verify_response(response)
 
-        return TypeAdapter(list[Queue]).validate_json(response.content)
+        page = TypeAdapter(PaginatedTasksResponse | None).validate_json(response.content)
```

Comment:
If the limit is hard coded to 1, do we really need a paginated response, especially when we then return only the first Task on line 89

Reply:
This is specifically left in to make the new changes work with the current setup, of the task runner handling one task at a time. The endpoint now allows for batched tasks to be popped from the endpoint as was requested in feature requests for the scheduler. I made @Donnype aware of these changes and he will be thinking about the necessary changes to the task runner to leverage this new functionality.

Reply:
However, come to think about it I need to check whether pagination will work correctly with tasks being popped and set to dispatched correctly.

Reply:
Makes sense, I was thinking it might have been because of the multi-pop support. But, I'm guessing that would either introduce an extra param (limit?) or a new method?

Reply:
popping and changing the status (for any number of taskst) should be handled in the same database-transaction I'd say.  Hopefully relying on table locking to ensure a task is not popped twice simultaneously.